app/src/main/java/com/github/mateo762/myapplication/ui/home/TodayScreen.kt

```diff
@@ -0,0 +1,237 @@
+        }
+        ?.second
+
+    val nextUpLocalDate = if (nextUpHabitDay != null) {
+        time.toLocalDate().plusDays(
+            if (nextUpHabitDay.ordinal >= today.ordinal) {
+                nextUpHabitDay.ordinal - today.ordinal
+            } else {
+                7 - (today.ordinal - nextUpHabitDay.ordinal)
+            }.toLong()
+        )
+    } else {
+        time.toLocalDate()
+    }
+
+    val nextUpLocalDateTime = if (nextUpHabit != null) {
+        LocalDateTime.of(nextUpLocalDate, parseTime(nextUpHabit.startTime))
+    } else {
+        time
+    }
+
+    val allocatedHours = Duration.between(time, nextUpLocalDateTime).toHours()
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .verticalScroll(rememberScrollState())
+            .padding(16.dp)
+    ) {
+        // Today
+        Box(
+            modifier = Modifier
+                .background(Color(0xFFFED4D6), RoundedCornerShape(8.dp))
+                .fillMaxWidth()
+                .padding(vertical = 16.dp, horizontal = 16.dp)
+        ) {
+            Column {
+                Text(
+                    text = "Today",
+                    style = MaterialTheme.typography.h3,
+                )
+                if (todayHabits.isEmpty()) {
+                    Text(
+                        text = "No habits due for today.",
+                        modifier = Modifier
+                            .padding(vertical = 8.dp, horizontal = 16.dp)
+                            .testTag("today_empty")
+                    )
+                } else {
+                    todayHabits.forEach {
+                        val isCompleted = parseTime(it.startTime).isBefore(now)
+                        val testTag = if (isCompleted) {
+                            "today_completed_${it.name}"
+                        } else {
+                            "today_pending_${it.name}"
+                        }
+                        Row(
+                            verticalAlignment = Alignment.CenterVertically,
+                            horizontalArrangement = Arrangement.SpaceBetween,
+                            modifier = Modifier
+                                .fillMaxWidth()
+                                .padding(vertical = 8.dp, horizontal = 16.dp)
+                                .testTag(testTag)
+                        ) {
+                            Text(
+                                text = it.name,
+                                style = if (isCompleted) {
+                                    MaterialTheme.typography.h5.copy(
+                                        textDecoration = TextDecoration.LineThrough
+                                    )
+                                } else {
+                                    MaterialTheme.typography.h5
+                                }
+                            )
+                            Text(
+                                text = "0\uD83D\uDD25",
```

Comment:
The idea of the flame emoji was as a marker of an ongoing habit that is being consistently completed, but maybe we can modify that later, once the live data is usable and the values can be updated

Reply:
Otherwise, looks good to me